### PR TITLE
11.0 add consumption mto rule

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web#id=13936&view_type=form&model=project.task&menu_id=200

### CODE REVIEW

File: project_wip_material/tests/test_task_picking_smart_button.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from ddt import ddt, data
+from odoo import fields
```
Comment (ddufresne) [inline project_wip_material/tests/test_task_picking_smart_button.py]: Codacy Issue found: [Unused fields imported from odoo](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4176999)

File: project_wip_material/tests/test_task_picking_smart_button.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline project_wip_material/tests/test_task_picking_smart_button.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4176999)

File: project_wip_material/consumption_route.py [hunk 3]
```diff
@@ -216,6 +221,39 @@ def _create_or_update_consumption_route(self):
...
+    def _create_consumption_mto_pull(self):
+        vals = self._get_consumption_mto_pull_vals()
+        self.consu_mto_pull_id = self.env['procurement.rule'].create(vals)
+
+    def _update_consumption_mto_pull(self):
+        vals = self._get_consumption_mto_pull_vals()
+        self.consu_mto_pull_id.write(vals)
+
+    def _create_or_update_consumption_mto_pull(self):
+        vals = self._get_consumption_mto_pull_vals()
```
Comment (ddufresne) [inline project_wip_material/consumption_route.py]: Codacy Issue found: [Unused variable 'vals'](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4176999)

Comment (foutoucour) [inline project_wip_material/consumption_route.py]:
```suggestion
```

File: project_wip_material/tests/test_task_picking_smart_button.py (hunk 1 above)
Comment (foutoucour) [reply project_wip_material/tests/test_task_picking_smart_button.py]:
```suggestion
```

File: project_wip_material/tests/test_task_picking_smart_button.py (hunk 2 above)
Comment (foutoucour) [reply project_wip_material/tests/test_task_picking_smart_button.py]:
```suggestion
```

**Review by foutoucour — APPROVED** [review]

File: project_wip_material/tests/test_task_picking_smart_button.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# © 2019 Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from ddt import ddt, data
+from odoo.exceptions import ValidationError
```
Comment (ddufresne) [inline project_wip_material/tests/test_task_picking_smart_button.py]: Codacy Issue found: [Unused ValidationError imported from odoo.exceptions](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=4176999)

**Pull request merged**